lumicks/pylake/population/dwelltime.py

```diff
@@ -744,7 +788,11 @@ def hist(
         component_kwargs = {"marker": "o", "ms": 3, **(component_kwargs or {})}
         fit_kwargs = {"color": "k", **(fit_kwargs or {})}
 
-        components = self.pdf(centers)
+        # We average over a relatively fine mesh to make sure our PDF estimates are good when not
+        # all t_min and t_max are the same.
+        n_sample = max(3, int_steps // n_bins)
+        time_points = np.hstack([np.linspace(t1, t2, n_sample) for t1, t2 in zip(bins, bins[1:])])
+        components = np.mean(self.pdf(time_points).reshape(self.n_components, -1, n_sample), axis=2)
```

Comment:
if you go with the suggestion of implementing a separate discretized PDF, could potentially wrap this in there too to keep the full implementation together

Reply:
I split the discretized PDF into its own function. While doing this, I actually noticed that the number of bins requested is not actually the number of bins you get. I fixed this small issue in a separate commit.